src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -498,16 +533,19 @@ Transactor::checkSingleSign(PreclaimContext const& ctx)
 {
     // Check that the value in the signing key slot is a public key.
     auto const pkSigner = ctx.tx.getSigningPubKey();
-    if (!publicKeyType(makeSlice(pkSigner)))
+    if (!(ctx.flags & tapDRY_RUN) && !publicKeyType(makeSlice(pkSigner)))
```

Comment:
Why there is no `ctx.flags & tapDRY_RUN` for `checkMultiSign` ?

Reply:
Added in 0b818f33287094da7e46370c20269dac49b7a540